pkg/clusters/addons/kong/kong.go

```diff
@@ -146,11 +167,44 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		)
 	}
 
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
 	// do the deployment and install the chart
 	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
-	args = append(args, "--create-namespace", "--namespace", a.namespace)
-	args = append(args, a.deployArgs...)
+	args = append(args, "--namespace", a.namespace)
+	if a.enterprise {
+		if a.license != "" && a.license == KongEnterpriseTestLicense {
+			if err := deployKongEnterpriseLicenseSecret(ctx, cluster, a.namespace, KongEnterpriseTestLicense); err != nil {
+				return err
+			}
+		}
+
+		if err := prepareSecrets(ctx, a.namespace); err != nil {
+			return err
+		}
+
+		args = append(args, "--version", "2.3.0", "-f", "https://raw.githubusercontent.com/Kong/charts/main/charts/kong/example-values/minimal-k4k8s-with-kong-enterprise.yaml")
```

Comment:
Using a specific version of the chart _plus_ using a `main` version of the examples seems problematic. If the examples change in `main` in a way that's backwards incompatible with the hardcoded version it would break this code retroactively.

I recommend we simply pull in any of the enterprise configuration we need and add it to the `--set` methods here.

Reply:
we agreed on keeping the version as it is, and create follow up to convert all parameters into helm charts parameters